pkg/client/retrieve.go

```diff
@@ -130,17 +134,15 @@ func UntarAll(reader io.Reader, destDir, filename string) (filenames []string, r
 	// If we fail to read them all then any pipes we are connected to
 	// may hang.
 	defer func() {
-		b, err := ioutil.ReadAll(reader)
-		if err != nil {
-			returnErr = errors.Wrap(err, "failed to drain the tar reader")
+		err := drainReader(reader)
+		switch {
+		case err == nil:
+			return
+		case returnErr == nil && err != nil:
+			returnErr = err
+		case returnErr != nil && err != nil:
+			logrus.Warnf("Error while draining reader (%v) but there was a previous error that may be the root cause: %v", err, returnErr)
```

Comment:
This section was just refactored for clarity when hitting both errors. Found when debugging the original issue so I included it here.